src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests/Functions/ProcessCompletionOfNextDataSetVersionImportFunctionTests.cs

```diff
@@ -0,0 +1,184 @@
+            ];
+
+            foreach (var activityName in expectedActivitySequence)
+            {
+                mockOrchestrationContext
+                    .InSequence(activitySequence)
+                    .Setup(context => context.CallActivityAsync(activityName,
+                        mockOrchestrationContext.Object.InstanceId,
+                        null))
+                    .Returns(Task.CompletedTask);
+            }
+
+            await ProcessCompletionOfNextDataSetVersionImport(mockOrchestrationContext.Object);
+
+            VerifyAllMocks(mockOrchestrationContext);
+        }
+
+        [Fact]
+        public async Task ActivityFunctionThrowsException_CallsHandleFailureActivity()
+        {
+            var mockOrchestrationContext = DefaultMockOrchestrationContext();
+
+            var activitySequence = new MockSequence();
+
+            mockOrchestrationContext
+                .InSequence(activitySequence)
+                .Setup(context =>
+                    context.CallActivityAsync(ActivityNames.ImportData,
+                        mockOrchestrationContext.Object.InstanceId,
+                        null))
+                .Throws<Exception>();
+
+            mockOrchestrationContext
+                .InSequence(activitySequence)
+                .Setup(context =>
+                    context.CallActivityAsync(ActivityNames.HandleProcessingFailure,
+                        mockOrchestrationContext.Object.InstanceId,
+                        null))
+                .Returns(Task.CompletedTask);
+
+            await ProcessCompletionOfNextDataSetVersionImport(mockOrchestrationContext.Object);
+
+            VerifyAllMocks(mockOrchestrationContext);
+        }
+
+        private async Task ProcessCompletionOfNextDataSetVersionImport(TaskOrchestrationContext orchestrationContext)
+        {
+            var function = GetRequiredService<ProcessCompletionOfNextDataSetVersionFunction>();
+            await function.ProcessCompletionOfNextDataSetVersion(
+                orchestrationContext,
+                new ProcessDataSetVersionContext { DataSetVersionId = Guid.NewGuid() });
+        }
+
+        private static Mock<TaskOrchestrationContext> DefaultMockOrchestrationContext(
+            Guid? instanceId = null,
+            bool isReplaying = false)
+        {
+            var mock = new Mock<TaskOrchestrationContext>(MockBehavior.Strict);
+
+            mock
+                .Protected()
+                .Setup<ILoggerFactory>("LoggerFactory")
+                .Returns(NullLoggerFactory.Instance);
+
+            mock
+                .SetupGet(context => context.InstanceId)
+                .Returns(instanceId?.ToString() ?? Guid.NewGuid().ToString());
+
+            mock
+                .SetupGet(context => context.IsReplaying)
+                .Returns(isReplaying);
+
+            return mock;
+        }
+    }
+
+    public class CompleteInitialDataSetVersionProcessingTests(
```

Comment:
I think these tests should be in `ProcessInitialDataSetVersionFunctionTests.cs`?

Reply:
Haha nope, this was supposed to be `CompleteNextDataSetVersionImportProcessingTests` testing the completion stage from `ProcessCompletionOfNextDataSetVersionFunction`.

I have now updated thanks!